### PR TITLE
Solution Exercise_05

### PR DESCRIPTION
**Pull request by CorvinAn97**
Part 1: 1639024365
Part 2: 6260

### CODE REVIEW

File: Exercises/Corvin Ansorge/exercises_05.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# and much more deeply nested than the example above. 
+# For the actual inputs, how many bags are inside your single shiny gold bag?
+# As usual, please list the answer as part of the PR.
+
+def get_bag_dict(fp):
+    with open(fp, 'r') as file:
+        bagdict = {}
+        for line in file:
+            key = line.split("contain")[0].replace(" ","").replace("bags","bag")
+            value = line.split("contain")[1].replace(".\n", "").replace(" ","").replace("bags","bag")
```
Comment (s231365) [inline Exercises/Corvin Ansorge/exercises_05.py:65]: it runs, but your multiple use of .replace could be solved better

File: Exercises/Corvin Ansorge/exercises_05.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        count = 0
+        for innerbag in bagdict[bag].split(","):
+            count += int(innerbag[0]) + int(innerbag[0]) * get_bag_count(bagdict, innerbag[1:])
+        return count
+
+def part2_solution(fp):
+    bagdict = get_bag_dict(fp)
+    return get_bag_count(bagdict, "shinygoldbag")
+
+print(part2_solution("/Users/corvinansorge/PycharmProjects/ppp-2023/data/input_bags.txt"))
```
Comment (s231365) [inline Exercises/Corvin Ansorge/exercises_05.py:82]: same problem with the path as above

**Review by s231365 — COMMENTED** [review]
your solution works, but some minor issues

File: Exercises/Corvin Ansorge/exercises_05.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# Find the first number in this list which can not be expressed as a
+# sum of two numbers out of the 25 numbers before it.
+# Please make not of your result in the PR.
+
+def part1_read_input():
+    with open("/Users/corvinansorge/PycharmProjects/ppp-2023/data/input_sequence.txt") as f:
+        return [int(line) for line in f.readlines()]
+
+def part1_solution(numbers, preamble_length):
+    for i in range(preamble_length, len(numbers)):
```
[the remark this answers is not in the document]
Comment (CorvinAn97) [reply Exercises/Corvin Ansorge/exercises_05.py:24]: sadly the truth. :s

File: Exercises/Corvin Ansorge/exercises_05.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+# numbers adds to 127.
+#
+# You can find the ACTUAL input for this exercise under `data/input_sequence.txt`. For this
+# real input you should consider not only the 5 numbers, but the 25 numbers preceding.
+# Find the first number in this list which can not be expressed as a
+# sum of two numbers out of the 25 numbers before it.
+# Please make not of your result in the PR.
+
+def part1_read_input():
+    with open("/Users/corvinansorge/PycharmProjects/ppp-2023/data/input_sequence.txt") as f:
```
[the remark this answers is not in the document]
Comment (CorvinAn97) [reply Exercises/Corvin Ansorge/exercises_05.py:20]: thanks for reminding me. Will fix that.

**Pull request merged**